### PR TITLE
Add SNIa rate plot and update SFR plot

### PR DESCRIPTION
**Pull request by Fonotec**
I added the SNIa rate plot as discussed last week and I also updated the SFR plot by adding more recent observations. This is how the plots look like now, in this simulation the SNIa rate is very finely logged here, so normally that should look more smooth if binned over longer times. 

Let me know if you have suggestions or comments. 

[SFH_Comparison.pdf](https://github.com/SWIFTSIM/swiftsimio-examples/files/4053644/SFH_Comparison.pdf)
[SNIa_rate_Comparison.pdf](https://github.com/SWIFTSIM/swiftsimio-examples/files/4053645/SNIa_rate_Comparison.pdf)

### CODE REVIEW

File: Supernovae_Ia/loadObservationalData.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+"""
+Code from Matthieu Schaller to load observational SFR data.
```
Comment (JBorrow) [inline Supernovae_Ia/loadObservationalData.py]: Should this be changed to something more realistic? I guess you wrote this!

File: star_formation_history/loadObservationalData.py [hunk 2]
```diff
@@ -100,7 +102,7 @@ def read_obs_data(path="observational_data"):
         )
     )
 
-    # SFR Observational data from Cucciati 2012
```
Comment (JBorrow) [inline star_formation_history/loadObservationalData.py]: ? Why this change

File: Supernovae_Ia/plotSNIaRate.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+ax.set_ylabel(r"SNIa rate $[\rm yr^{-1} \cdot Mpc^{-3}]$")
+
+
+redshift_ticks = np.array([0.0, 0.2, 0.5, 1.0, 2.0, 3.0, 5.0, 10.0, 20.0, 50.0, 100.0])
+redshift_labels = [
+    "$0$",
+    "$0.2$",
+    "$0.5$",
+    "$1$",
+    "$2$",
```
Comment (JBorrow) [inline Supernovae_Ia/plotSNIaRate.py:123]: Why do these redshift labels not work in your plot?

File: Supernovae_Ia/plotSNIaRate.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    "$50$",
+    "$100$",
+]
+a_ticks = 1.0 / (redshift_ticks + 1.0)
+
+ax.set_xticks(a_ticks)
+ax.set_xticklabels(redshift_labels)
+ax.tick_params(axis="x", which="minor", bottom=False)
+
+ax.set_xlim(1.02, 0.07)
```
Comment (JBorrow) [inline Supernovae_Ia/plotSNIaRate.py]: we should set this to a better default for the SN1A

File: star_formation_history/loadObservationalData.py [hunk 5]
```diff
@@ -63,7 +63,8 @@ def read_obs_data(path="observational_data"):
         ObservationalData(obs2_a, obs2_rho, obs2_rho_err, "Karim et al. (2011) [radio]")
     )
 
-    # SFR Observational data from Bouwens 2012
+    # SFR Observational data from Bouwens 2012 (not used included by Bouwens 2015)
```
Comment (JBorrow) [inline star_formation_history/loadObservationalData.py:66]: What does this comment mean?

**Review by JBorrow — CHANGES_REQUESTED** [review]
Looks good in general! Just a few comments.

File: star_formation_history/loadObservationalData.py (hunk 5 above)
Comment (Fonotec) [reply star_formation_history/loadObservationalData.py:66]: The Bouwens et al. (2012) observations are included and with additional data this results in the Bouwens et al. (2015) data points, so in this version I commented out the Bouwens et al. (2012) data points.

File: Supernovae_Ia/plotSNIaRate.py (hunk 3 above)
Comment (Fonotec) [reply Supernovae_Ia/plotSNIaRate.py:123]: Btw they work, I see that I added a figure in which I changed xscale to log, so than it is not included, but in the current version in this merge it should work.

**Comment by Fonotec** [comment]
The SNIa rate should look like this, I uploaded the wrong version. 

[SNIa_rate_Comparison.pdf](https://github.com/SWIFTSIM/swiftsimio-examples/files/4054252/SNIa_rate_Comparison.pdf)

File: Supernovae_Ia/loadObservationalData.py (hunk 1 above)
Comment (Fonotec) [reply Supernovae_Ia/loadObservationalData.py]: I changed this.

File: star_formation_history/loadObservationalData.py (hunk 2 above)
Comment (Fonotec) [reply star_formation_history/loadObservationalData.py]: I changed this back

File: Supernovae_Ia/plotSNIaRate.py (hunk 4 above)
Comment (Fonotec) [reply Supernovae_Ia/plotSNIaRate.py]: I think we cannot change the default a lot, given that the plot I show to you has a delay that only kicks in around redshift 3.5, but some models already have a high rate around 7/8, so I will only slightly change it to 0.10.

**Comment by JBorrow** [comment]
Thanks very much for this Folkert!
